File: app/src/main/java/com/kml/data/externalDbOperations/DbGetAllUsersData.java
package com.kml.data.externalDbOperations;

import android.util.Log;

import java.net.HttpURLConnection;

public class DbGetAllUsersData extends ExternalDbHelper
{
    private final String fileName = "getAllDataAboutUser.php";

    String address;
    String result;
    HttpURLConnection conn;

    public DbGetAllUsersData()
    {
        address = BASE_URL+fileName;
    }

    @Override
    public void run()
    {
        String dataFromDb;

        conn = setConnection(address);
        dataFromDb = readResult(conn);

        setResult(dataFromDb);
    }

    private void setResult(String result)
    {
        this.result = result;
    }

    public String getResult()
    {
        try {
            this.join();
        } catch (InterruptedException e) {
            Log.d("INTERRUPTED_EXCEPTION", "getResult: " + result + " -> " + e.getMessage());
        }
        Log.d("RESULT_GET_ALL", "getResult: "+result);
        return result;
    }
}

File: app/src/main/java/com/kml/data/externalDbOperations/DbGetAllUsersData.kt
package com.kml.data.externalDbOperations

import java.net.HttpURLConnection

class DbGetAllUsersData : ExternalDbHelper() {
    private val fileName = "getAllDataAboutUser.php"
    private var address = BASE_URL + fileName
    var result: String = ""
        get() {
            join(); return field
        }
    var conn: HttpURLConnection? = null

    override fun run() {

        conn = setConnection(address)
        val dataFromDb: String = readResult(conn!!)
        result = dataFromDb

    }
}